substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProviderBase.java

```diff
@@ -426,6 +444,22 @@ private static HashMap<JavaKind, HostedType> initJavaKindToHostedTypes(HostedMet
         return map;
     }
 
+    /**
+     * Retrieve details of the native calling convention for a top level compiled method, including
+     * details of which registers or stack slots are used to pass parameters.
+     * 
+     * @param method The method whose calling convention is required.
+     * @return The calling convention for the method.
+     */
+    protected SubstrateCallingConvention getCallingConvention(HostedMethod method) {
+        SubstrateCallingConventionKind callingConventionKind = method.getCallingConventionKind();
+        HostedType declaringClass = method.getDeclaringClass();
+        HostedType receiverType = method.isStatic() ? null : declaringClass;
+        Signature signature = method.getSignature();
+        SubstrateCallingConventionType type = callingConventionKind.toType(false);
+        return (SubstrateCallingConvention) registerConfig.getCallingConvention(type, signature.getReturnType(declaringClass), signature.toParameterTypes(receiverType), valueKindFactory);
```

Comment:
You can use `ResolvedJavaMethod.toParameterTypes()` to avoid dealing with `receiverType`.
```suggestion
        return (SubstrateCallingConvention) registerConfig.getCallingConvention(type, signature.getReturnType(declaringClass), method.toParameterTypes(), valueKindFactory);
```

Reply:
Calling `method.toParameterTypes()` fails when the method is a constructor. In that case the calling convention needs to take account of the 'this' parameter as well as the constructor arguments. However, if I were to call `method.toParameterTypes()` when `method` was a constructor it would call `method.getSignature().toParameterTypes(null)` and the resulting calling convention would omit the 'this' argument. That is why my code handles this case specially and only passes null when the method is static.